File: year2023/di-lena/pseudo-C-py/mcd-euclide/mcd.c
//algoritmo di euclide per massimo comune divisore tra x e y (x >= y)
#include <stdio.h>
#include <stdlib.h>
#include <libgen.h>

int mcd (int x, int y) {
	if (y == 0)
		return x;
	else {
		int r = x % y;
		return mcd(y,r);
	}
}
int main(int argc, char* argv[]) {
	if( argc != 3 ) {
		printf("usage: %s x y\n", argv[0]);
		return -1;
	}
	int a = atoi(argv[1]);
	int b = atoi(argv[2]);
	if( ! (a >= b) ) {
		printf("should value (x >= y)\n");
		return -1;
	}
	printf("mcd(%d,%d) = %d\n", a, b, mcd(a, b));
	return 0;
}

File: year2023/di-lena/pseudo-C-py/01-mcd-euclide/mcd.py
#!/bin/env/python3

#algoritmo di euclide per massimo comune divisore
#tra x e y (x >= y)

import sys;

def mcd (x, y):
	if (y == 0):
		return x;
	else:
		r = x % y;
		return mcd(y,r);

if __name__ == '__main__':
	try:
		a = sys.argv[1];
		b = sys.argv[2];
		print(f"mcd({a},{b}) = {mcd(a,b)}");
	except:
		print(f"usage: {sys.argv[0]} x y");
		print("should value (x >= y)");

